projects/packages/assets/src/class-script-data.php

```diff
@@ -86,17 +87,18 @@ public static function render_script_data() {
 		self::$did_render_script_data = true;
 
 		$script_data = is_admin() ? self::get_admin_script_data() : self::get_public_script_data();
-
-		$script_data = wp_json_encode(
+		$script_json = wp_json_encode(
 			$script_data,
 			JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP | JSON_UNESCAPED_UNICODE
 		);
 
-		wp_add_inline_script(
-			self::SCRIPT_HANDLE,
-			sprintf( 'window.JetpackScriptData = %s;', $script_data ),
-			'before'
-		);
+		if ( is_admin() || did_action( 'enqueue_block_editor_assets' ) ) {
+			wp_enqueue_script( self::SCRIPT_HANDLE );
+			wp_add_inline_script( self::SCRIPT_HANDLE, "window.JetpackScriptData = {$script_json};", 'before' );
+		} else {
+			// For public pages, we directly print the script tag.
+			printf( '<script>window.JetpackScriptData = %s;</script>', $script_json ); //phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- data encoded earlier.
```

Comment:
I ask the question above because you use a `printf` here instead of using the JSON data variable directly in the string. It seems weird to change existing code away from one way of adding data to a string just to use the old way in new code a few lines later.

It would be better to use the same way of modifying the string in both places.

Reply:
Yes, good question. For sure, we can be more consistent here. Originally I was working with one set of changes so missed an opportunity for more consistency here after changing away from `sprintf`.
Given the `sprintf` changes with PHP 8.4 (automatically rewriting to use string interpolation in cases like this) I think it's fine to stick with the `sprintf` / `printf` combination rather than string interpolation throughout. Seemingly minor differences anyway. Thanks for spotting this!

Reply:
Thanks. The changes look good. I'll test again to confirm that everything works as expected.